### DISCUSSION TITLE
Enhancement: Enable Web Search without Query Generation

### DISCUSSION THREAD

**Discussion by super-qua** [open-webui/open-webui#5385] [General]
**Is your feature request related to a problem? Please describe.**
I would like to be able to search with the Web Search, but use my own search query instead of one generated by an LLM.

**Describe the solution you'd like**
The two settings _Enable Web Search_ and _Enable Web Search Query Generation_ should be distinct from one another. 
The first one should enable the web search feature, while the second one should handle if the user query is passed to a LLM to generate the query for the search engine.

By enabling _Enable Web Search_ without _Enable Web Search Query Generation_, the user query should be passed directly to the search engine.

**Describe alternatives you've considered**
The only alternative at the moment would be to use a tool for this.

**Additional context**
The response should still be summarised by the LLM.
This feature would help to increase privacy options by not sending the query to a LLM provider for example.

**Comment by tjbck** [comment]
I believe this is already the case with the latest main,

<img width="1246" alt="image" src="https://github.com/user-attachments/assets/0623ca99-7583-4e67-83f6-ae4cd67a10bd">

**Comment by super-qua** [comment]
oh yes, you are right. Seems like I just had a UI mix-up with the toggles. my mistake, sorry for the noise.

**Comment by Magicen0722** [comment]
Do you have any methods to make the search query prompt effective? I customized the search query prompt and increased the threshold for generating prompt length, but it still only searches for the original question.
```
Assess the need for a web search based on the current question and prior interactions, but lean towards suggesting a Google search query if uncertain. Generate a Google search query even when the answer might be straightforward, as additional information may enhance comprehension or provide updated data. If absolutely certain that no further information is required, return an empty string. Default to a search query if unsure or in doubt. Today's date is {{CURRENT_DATE}}.

Current Question:
{{prompt:end:4000}}
```
![_17262971154644](https://github.com/user-attachments/assets/2e22d0c4-4837-4fa5-9b4c-e1de817140ac)

**Comment by skorokithakis** [comment]
Where can I find this setting in the UI? Even if I type "cinemas london", for example, I still get a search query of the LLM's stream of thought like "given the current terms, it looks like the user wants to search for cinemas in London", which ruins the search.

**Comment by skorokithakis** [comment]
It looks like something's wrong, the default for the ENABLE_SEARCH_QUERY variable isn't actually `false`. I set it to `false` explicitly, and now the query gets passed through.
